src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpContext.cs

```diff
@@ -374,159 +370,16 @@ public unsafe void SendResponseHeaders(bool appCompleted)
 
-            // We should be sending the headers here as there are many responses that don't have status codes.
-            _pinnedHeaders = SerializeHeaders(pHttpResponse);
-        }
-
-        // From HttpSys, except does not write to response
-        private unsafe List<GCHandle> SerializeHeaders(HttpApiTypes.HTTP_RESPONSE_V2* pHttpResponse)
-        {
-            HttpResponseHeaders.IsReadOnly = true;
-            HttpApiTypes.HTTP_UNKNOWN_HEADER[] unknownHeaders = null;
-            HttpApiTypes.HTTP_RESPONSE_INFO[] knownHeaderInfo = null;
-            var pinnedHeaders = new List<GCHandle>();
-            GCHandle gcHandle;
-
-            if (HttpResponseHeaders.Count == 0)
-            {
-                return null;
-            }
-            string headerName;
-            string headerValue;
-            int lookup;
-            var numUnknownHeaders = 0;
-            int numKnownMultiHeaders = 0;
-            byte[] bytes = null;
-
             foreach (var headerPair in HttpResponseHeaders)
             {
-                if (headerPair.Value.Count == 0)
-                {
-                    continue;
-                }
-                lookup = HttpApiTypes.HTTP_RESPONSE_HEADER_ID.IndexOfKnownHeader(headerPair.Key);
-                if (lookup == -1) // TODO handle opaque stream upgrade?
-                {
-                    numUnknownHeaders++;
-                }
-                else if (headerPair.Value.Count > 1)
-                {
-                    numKnownMultiHeaders++;
-                }
-            }
-
-            try
-            {
-                var pKnownHeaders = &pHttpResponse->Response_V1.Headers.KnownHeaders;
-                foreach (var headerPair in HttpResponseHeaders)
-                {
-                    if (headerPair.Value.Count == 0)
-                    {
-                        continue;
-                    }
-                    headerName = headerPair.Key;
-                    StringValues headerValues = headerPair.Value;
-                    lookup = HttpApiTypes.HTTP_RESPONSE_HEADER_ID.IndexOfKnownHeader(headerName);
-                    if (lookup == -1)
-                    {
-                        if (unknownHeaders == null)
-                        {
-                            unknownHeaders = new HttpApiTypes.HTTP_UNKNOWN_HEADER[numUnknownHeaders];
-                            gcHandle = GCHandle.Alloc(unknownHeaders, GCHandleType.Pinned);
-                            pinnedHeaders.Add(gcHandle);
-                            pHttpResponse->Response_V1.Headers.pUnknownHeaders = (HttpApiTypes.HTTP_UNKNOWN_HEADER*)gcHandle.AddrOfPinnedObject();
-                            pHttpResponse->Response_V1.Headers.UnknownHeaderCount = 0; // to remove the iis header for server=...
-                        }
+                var headerNameBytes = Encoding.UTF8.GetBytes(headerPair.Key);
+                var headerValueBytes = Encoding.UTF8.GetBytes(headerPair.Value);
```

Comment:
Any way to make IIS send multi-line headers rather than squashing them together? E.g.
HeaderName1: Value1
HeaderName1: Value2
...
This is legal, and required by some headers like auth and set-cookie.

Reply:
I think we will need to check for multilineheaders here. IIS will write them as a comma separated header. Do we want to remove any existing IIS headers? If not, we can always set fReplace to false.